codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/compression/PerMessageDeflateEncoder.java

```diff
@@ -63,7 +64,11 @@ protected boolean removeFrameTail(WebSocketFrame msg) {
     @Override
     protected void encode(ChannelHandlerContext ctx, WebSocketFrame msg,
             List<Object> out) throws Exception {
-        super.encode(ctx, msg, out);
+        if (msg instanceof UncompressedWebSocketFrame) {
+            out.add(((UncompressedWebSocketFrame) msg).unwrap());
```

Comment:
@normanmaurer I'm not 100% sure, perhaps it's better to retain the msg here instead of retain the msg here?

```
msg.retain();
out.add(((UncompressedWebSocketFrame) msg).unwrap());
```